test/extended/networking/network_segmentation.go

```diff
@@ -901,15 +901,20 @@ func applyManifest(namespace, manifest string) error {
 }
 
 func waitForUserDefinedNetworkReady(namespace, name string, timeout time.Duration) error {
-	_, errNetReady := e2ekubectl.RunKubectl(namespace, "wait", "userdefinednetwork", name, "--for", "condition=NetworkReady=True", "--timeout", timeout.String())
-	netReady := errNetReady == nil
-	_, errNetCreated := e2ekubectl.RunKubectl(namespace, "wait", "userdefinednetwork", name, "--for", "condition=NetworkCreated=True", "--timeout", timeout.String())
+	// kubectl wait for condition assumes the resource exists
+	// in CI sometimes it could take a while for the reasource to appear in etcd so better to first
+	// wait for the resource to be created before attempting to check the status
+	_, errNetCreated := e2ekubectl.RunKubectl(namespace, "wait", "userdefinednetwork", name, "--for=create", "--timeout", timeout.String())
```

Comment:
I think it should fail early in case the object has not been created by the timeout, WDYT?

Reply:
hmm what do you mean? you mean to not make this condition a part of this util and put it outside in the test itself?

Reply:
I mean return error in case the object is not created on time.
If it created, check it has of the expected conditions.
```
_, errNetCreated := e2ekubectl.RunKubectl(namespace, "wait", "userdefinednetwork", name, "--for=create", "--timeout", timeout.String())
if errNetCreated != nil {
  return errNetCreated
}
...
```

Reply:
done.